File: src/main/java/architecture/ego_equipment/common/item/weapon/remote/special/CherryBlossomsWeaponItem.java
package architecture.ego_equipment.common.item.weapon.remote.special;

import architecture.ego_equipment.common.item.weapon.remote.RemoteEgoWeaponGeoItem;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.entity.LivingEntity;
import net.minecraft.world.entity.projectile.Projectile;
import org.jetbrains.annotations.Nullable;
import software.bernie.geckolib.model.GeoModel;

public class CherryBlossomsWeaponItem extends RemoteEgoWeaponGeoItem {

	public CherryBlossomsWeaponItem(Properties itemProperties, Builder egoWeaponBuilder, GeoModel<RemoteEgoWeaponGeoItem> geoModel, GeoModel<RemoteEgoWeaponGeoItem> guiModel) {
		super(itemProperties, egoWeaponBuilder, geoModel, guiModel);
	}

	public CherryBlossomsWeaponItem(Properties itemProperties, Builder egoWeaponBuilder, ResourceLocation modPath) {
		super(itemProperties, egoWeaponBuilder, modPath);
	}


	@Override
	protected void shootProjectile(LivingEntity shooterEntity, Projectile projectileEntity, int projectileIndex, float projectileVelocity, float projectileInaccuracy, float shootingAngle, @Nullable LivingEntity targetEntity) {

	}
}

File: src/main/kotlin/architecture/ego_equipment/common/item/weapon/remote/special/CherryBlossomsWeaponItem.kt
package architecture.ego_equipment.common.item.weapon.remote.special

import architecture.ego_equipment.common.item.weapon.remote.RemoteEgoWeaponGeoItem
import net.minecraft.resources.ResourceLocation
import net.minecraft.world.entity.LivingEntity
import net.minecraft.world.entity.projectile.Projectile
import software.bernie.geckolib.model.GeoModel

class CherryBlossomsWeaponItem : RemoteEgoWeaponGeoItem {
	constructor(
		itemProperties: Properties,
		egoWeaponBuilder: Builder,
		geoModel: GeoModel<RemoteEgoWeaponGeoItem>,
		guiModel: GeoModel<RemoteEgoWeaponGeoItem>?
	) : super(itemProperties, egoWeaponBuilder, geoModel, guiModel)

	constructor(
		itemProperties: Properties,
		egoWeaponBuilder: Builder,
		modPath: ResourceLocation
	) : super(itemProperties, egoWeaponBuilder, modPath)

	override fun shootProjectile(
		shooterEntity: LivingEntity, projectileEntity: Projectile, projectileIndex: Int,
		projectileVelocity: Float, projectileInaccuracy: Float, shootingAngle: Float, targetEntity: LivingEntity?
	) {
	}
}
